### PR TITLE
lms check if can edit course

### PR DESCRIPTION
**Pull request by keith-ferney**

### CODE REVIEW

File: app/controllers/documents_controller.rb [hunk 1]
```diff
@@ -254,7 +253,20 @@ def update
 
   protected
   def can_use_edit_token
-    if session[:lms_authenticated_user] || @organization[:enable_anonymous_actions] || has_role("designer")
+    lms_connection_information
```
Comment (llamapope) [inline app/controllers/documents_controller.rb]: Move lms_course_id to an argument, pass in when relevant
Clean up these checks, should do the least amount of work first (simple boolean checks), then do the canvas check last
Move lms session checks to helper method
Move canvas code to seperate method as well

**Comment by keith-ferney** [comment]
changes applied

File: app/controllers/documents_controller.rb [hunk 2]
```diff
@@ -253,8 +252,49 @@ def update
...
+      elsif authorized_to_edit_course(lms_course_id)
+        true
+      end
+    end
+  end
+
+  def authorized_to_edit_course lms_course_id
+    lms_connection_information
+    courses = get_canvas_courses
+    if courses.pluck("id").reject { |a| a != lms_course_id.to_i }.length < 1
```
Comment (llamapope) [inline app/controllers/documents_controller.rb]: simplify this, use include or in

File: app/controllers/documents_controller.rb [hunk 3]
```diff
@@ -253,8 +252,50 @@ def update
...
+      true
+    elsif is_lms_authenticated_user? && has_canvas_access_token? && lms_course_id
+      if @document == nil
+        true
+      elsif lms_course_id == nil
+        true
+      elsif authorized_to_edit_course(lms_course_id)
+        true
+      end
+    end
```
Comment (llamapope) [inline app/controllers/documents_controller.rb]: Don't we need a return of false at the end here? If they aren't logged in, but there is a course_id

File: app/controllers/documents_controller.rb [hunk 4]
```diff
@@ -253,8 +252,50 @@ def update
...
+  end
+
+  def authorized_to_edit_course lms_course_id
+    courses = get_canvas_courses
+    if courses.pluck("id").reject { |a| a != lms_course_id.to_i }.length < 1
```
Comment (llamapope) [inline app/controllers/documents_controller.rb]: I still want this simplified

**Comment by keith-ferney** [comment]
added false for not logged in and there is a lms_course_id and already simplified authorized_to_edit_course just forgot to push it

**Pull request merged**